src/components/timeRangePicker/TimeRangePicker.tsx

```diff
@@ -0,0 +1,302 @@
+          />
+          <Datepicker>
+            {isCalendarActive && (
+              <Calendar
+                className="calendar-panel"
+                date={selectedDate}
+                weekStartsOn={1}
+                onChange={(date) => handleDateSelection(date)}
+              />
+            )}
+          </Datepicker>
+        </DatepickerContainer>
+      </Row>
+      <Row className="time-range-picker">
+        <Label name="Hours:" />
+        <UnifiedMultipleSelect size="small">
+          {DEFAULT_TIME_RANGE_OPTIONS.map((timeRangeOption) => (
+            <SelectButton
+              key={timeRangeOption.id}
+              name={timeRangeOption.id}
+              value={timeRangeOption.id}
+              labelText={`${timeRangeOption.start} - ${timeRangeOption.end}`}
+              onClick={() => setSelectedTimeRange(() => timeRangeOption)}
+              isActive={timeRangeOption.id === selectedTimeRange.id}
+            />
+          ))}
+          <SelectButton
+            key="custom"
+            name="custom"
+            value="custom"
+            labelText="custom"
+            onClick={() =>
+              setSelectedTimeRange((prev) => {
+                return {
+                  ...prev,
+                  id: "custom",
+                };
+              })
+            }
+            isActive={selectedTimeRange.id === "custom"}
+          />
+        </UnifiedMultipleSelect>
+        <CustomTimeRangeSelector isVisible={showCustomTimeRange}>
+          <div>From</div>
+          <input
+            type="time"
+            name="time-range-start"
+            value={selectedTimeRange.start}
+            onChange={(e) => {
+              e.persist();
+              setSelectedTimeRange((prev) => {
+                return {
+                  ...prev,
+                  start: e.target.value,
+                };
+              });
+            }}
+          />
+          <div>to</div>
+          <input
+            type="time"
+            name="time-range-end"
+            value={selectedTimeRange.end}
+            min={selectedTimeRange.start}
+            onChange={(e) => {
+              e.persist();
+              setSelectedTimeRange((prev) => {
+                return {
+                  ...prev,
+                  end: e.target.value,
+                };
+              });
+            }}
+          />
+          <Button severity="high" onClick={submit} size="sm">
+            Apply
+          </Button>
+        </CustomTimeRangeSelector>
+      </Row>
+    </>
+  );
+}
+
+const getMonday = (date: Date): Date => {
+  const result = new Date(date);
+
+  if (isMonday(result)) return result;
+  else return previousMonday(result);
+};
+
+const getInitSelectedDate = (initRange: Interval): Date => {
+  let initDate: Date;
+
+  if (isSameDay(initRange.start, initRange.end)) {
+    initDate = new Date(initRange.start);
+  } else {
+    initDate = new Date();
+  }
+
+  return initDate;
```

Comment:
ternary would simplify things here

```suggestion
  return isSameDay(initRange.start, initRange.end) ? new Date(initRange.start) : initDate;
```